pkg/container/qemu_runner.go

```diff
@@ -0,0 +1,761 @@
+		User: user,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+		Config: ssh.Config{
+			Ciphers: []string{"aes128-ctr"},
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(), // equivalent to StrictHostKeyChecking=no
+	}
+
+	// Connect to the SSH server
+	client, err := ssh.Dial("tcp", address, config)
+	if err != nil {
+		clog.FromContext(ctx).Errorf("Failed to dial: %s", err)
+		return err
+	}
+	defer client.Close()
+
+	// Create a session
+	session, err := client.NewSession()
+	if err != nil {
+		clog.FromContext(ctx).Errorf("Failed to create session: %s", err)
+		return err
+	}
+	defer session.Close()
+
+	for k, v := range cfg.Environment {
+		err = session.Setenv(k, v)
+		if err != nil {
+			return err
+		}
+	}
+
+	for k, v := range extraVars {
+		err = session.Setenv(k, v)
+		if err != nil {
+			return err
+		}
+	}
+
+	session.Stdin = stdin
+	session.Stderr = stderr
+	session.Stdout = stdout
+
+	b64cmd := base64.StdEncoding.EncodeToString([]byte(shellescape.QuoteCommand(command)))
+	cmd := strings.Join(
+		[]string{
+			"_c=$(echo " + b64cmd + "| base64 -d) || exit 97",
+			"eval set -- \"$_c\" || exit 98",
+			"exec \"$@\"",
+		}, " ;")
+
+	clog.FromContext(ctx).Infof("running (%d) %v", len(command), command)
+	err = session.Run(cmd)
+	if err != nil {
+		clog.FromContext(ctx).Errorf("Failed to run command %v: %s", command, err)
+		return err
+	}
+
+	return nil
+}
+
+func generateSSHKeys(ctx context.Context, cfg *Config) ([]byte, error) {
+	clog.FromContext(ctx).Info("qemu: generating ssh key pairs for ephemeral VM")
+	// Private Key generation
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get ASN.1 DER format
+	privDER, err := x509.MarshalECPrivateKey(privateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// pem.Block
+	privBlock := pem.Block{
+		Type:    "EC PRIVATE KEY",
+		Headers: nil,
+		Bytes:   privDER,
+	}
+
+	// Private key in PEM format
+	cfg.SSHKey = pem.EncodeToMemory(&privBlock)
+
+	publicKey, err := ssh.NewPublicKey(&privateKey.PublicKey)
+	if err != nil {
+		clog.FromContext(ctx).Errorf("qemu: ssh keygen failed: %v", err)
+		return nil, err
+	}
+
+	return ssh.MarshalAuthorizedKey(publicKey), nil
+}
+
+func convertHumanToKB(memory string) (int64, error) {
+	// Map of unit multipliers
+	unitMultipliers := map[string]int64{
+		"Ki": 1024,                      // Kibibytes
+		"Mi": 1024 * 1024,               // Mebibytes
+		"Gi": 1024 * 1024 * 1024,        // Gibibytes
+		"Ti": 1024 * 1024 * 1024 * 1024, // Tebibytes
+		"K":  1024,                      // Kilobytes (KB)
+		"M":  1024 * 1024,               // Megabytes (MB)
+		"G":  1024 * 1024 * 1024,        // Gigabytes (GB)
+		"T":  1024 * 1024 * 1024 * 1024, // Terabytes (TB)
+		"k":  1024,                      // Kilobytes (KB)
+		"m":  1024 * 1024,               // Megabytes (MB)
+		"g":  1024 * 1024 * 1024,        // Gigabytes (GB)
+		"t":  1024 * 1024 * 1024 * 1024, // Terabytes (TB)
```

Comment:
oh yeah, you should fix the comments or the numbers.
Either comment 'Kibibytes' or change it to 1000.